Detour/Include/DetourNavMesh.h

```diff
@@ -162,6 +171,20 @@ struct dtPoly
 	/// The user defined polygon flags.
 	unsigned short flags;
 
+#ifdef NAVMESH_ISLAND_SYSTEM
+	/// The island index.
+	unsigned int islandIdx;
+
+	/// Gets island index
+	/// @return The island index.
+	inline unsigned int getIslandIdx() const { return islandIdx; }
+
+	/// Set island index
+	///  @param[in]	idx		The new island index.
+	inline void setIslandIdx(unsigned int idx) { islandIdx = idx; }
+
+#endif
```

Comment:
Instead of modifying `dtPoly`, we can simply store data per poly by using an array. You can check the nav mesh purge tool on how to do that:
https://github.com/recastnavigation/recastnavigation/blob/9f9efe943ed54a31b02270fb13b301754e918528/RecastDemo/Source/NavMeshPruneTool.cpp#L40-L131

The benefit of this is that folks do not pay for what they don't use, and you can get rid of `NAVMESH_ISLAND_SYSTEM` as a bonus.

Reply:
done